sdks/java/io/amazon-web-services/src/test/java/org/apache/beam/sdk/io/aws/dynamodb/DynamoDBConfigMock.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.aws.dynamodb;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.options.ValueProvider;
+
+/** Mock DynamoDBConfiguration to pass it over the write to writer for testing Writer's retries. */
+public class DynamoDBConfigMock extends DynamoDBIO.DynamoDBConfiguration {
```

Comment:
Can you please remove this class and do the configuration of the AmazonDynamoDB class by following an approach like the one on KinesisIO: Two methods: `withAWSClientsProvider(AWSClientsProvider awsClientsProvider)` and `withAWSClientsProvider(
        String awsAccessKey, String awsSecretKey, Regions region, String serviceEndpoint)`. `AWSClientsProvider` can be like this:

```java
public interface AWSClientsProvider extends Serializable {
  AmazonDynamoDB getAmazonDynamoDB();
}
```

Reply:
Sure, will get back that AWSClientProvider. But this Configuration approach is copied from JDBC IO one

Reply:
Yes other IOs do it like that but for Amazon is probably a good idea to keep it similar, thanks!
A future idea we have is to make it configurable via options because users mostly already will have `AwsOptions` around, but do not do this for this PR we have to discuss this first in the mailing list.